cpp/src/plasma/plasma_extension.cc

```diff
@@ -0,0 +1,464 @@
+  Py_RETURN_NONE;
+}
+
+PyObject *PyPlasma_release(PyObject *self, PyObject *args) {
+  PlasmaConnection *conn;
+  ObjectID object_id;
+  if (!PyArg_ParseTuple(args, "O&O&", PyObjectToPlasmaConnection, &conn,
+                        PyStringToUniqueID, &object_id)) {
+    return NULL;
+  }
+  plasma_release(conn, object_id);
+  Py_RETURN_NONE;
+}
+
+PyObject *PyPlasma_get(PyObject *self, PyObject *args) {
+  PlasmaConnection *conn;
+  PyObject *object_id_list;
+  long long timeout_ms;
+  if (!PyArg_ParseTuple(args, "O&OL", PyObjectToPlasmaConnection, &conn,
+                        &object_id_list, &timeout_ms)) {
+    return NULL;
+  }
+
+  Py_ssize_t num_object_ids = PyList_Size(object_id_list);
+  ObjectID *object_ids = (ObjectID *) malloc(sizeof(ObjectID) * num_object_ids);
+  ObjectBuffer *object_buffers =
+      (ObjectBuffer *) malloc(sizeof(ObjectBuffer) * num_object_ids);
+
+  for (int i = 0; i < num_object_ids; ++i) {
+    PyStringToUniqueID(PyList_GetItem(object_id_list, i), &object_ids[i]);
+  }
+
+  Py_BEGIN_ALLOW_THREADS;
+  plasma_get(conn, object_ids, num_object_ids, timeout_ms, object_buffers);
+  Py_END_ALLOW_THREADS;
+  free(object_ids);
+
+  PyObject *returns = PyList_New(num_object_ids);
+  for (int i = 0; i < num_object_ids; ++i) {
+    if (object_buffers[i].data_size != -1) {
+      /* The object was retrieved, so return the object. */
+      PyObject *t = PyTuple_New(2);
+#if PY_MAJOR_VERSION >= 3
+      PyTuple_SetItem(
+          t, 0, PyMemoryView_FromMemory(
+                    (char *) object_buffers[i].data,
+                    (Py_ssize_t) object_buffers[i].data_size, PyBUF_READ));
+      PyTuple_SetItem(
+          t, 1, PyMemoryView_FromMemory(
+                    (char *) object_buffers[i].metadata,
+                    (Py_ssize_t) object_buffers[i].metadata_size, PyBUF_READ));
+#else
+      PyTuple_SetItem(
+          t, 0, PyBuffer_FromMemory((void *) object_buffers[i].data,
+                                    (Py_ssize_t) object_buffers[i].data_size));
+      PyTuple_SetItem(t, 1, PyBuffer_FromMemory(
+                                (void *) object_buffers[i].metadata,
+                                (Py_ssize_t) object_buffers[i].metadata_size));
+#endif
+      PyList_SetItem(returns, i, t);
+    } else {
+      /* The object was not retrieved, so just add None to the list of return
+       * values. */
+      Py_XINCREF(Py_None);
+      PyList_SetItem(returns, i, Py_None);
+    }
+  }
+  free(object_buffers);
+  return returns;
+}
+
+PyObject *PyPlasma_contains(PyObject *self, PyObject *args) {
+  PlasmaConnection *conn;
+  ObjectID object_id;
+  if (!PyArg_ParseTuple(args, "O&O&", PyObjectToPlasmaConnection, &conn,
+                        PyStringToUniqueID, &object_id)) {
+    return NULL;
+  }
+  int has_object;
+  plasma_contains(conn, object_id, &has_object);
+
+  if (has_object)
+    Py_RETURN_TRUE;
+  else
+    Py_RETURN_FALSE;
+}
+
+PyObject *PyPlasma_fetch(PyObject *self, PyObject *args) {
```

Comment:
I assume we're moving the plasma store and not the plasma manager to Arrow right now, so some of these won't actually compile and should be removed (`PyPlasma_fetch`, `PyPlasma_wait`). Also `PyPlasma_connect` will have to be changed to work without a manager.